client/components/data/query-post-counts/index.jsx

```diff
@@ -1,58 +1,35 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestPostCounts } from 'state/posts/counts/actions';
 import { isRequestingPostCounts } from 'state/posts/counts/selectors';
 
-class QueryPostCounts extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId && this.props.type === nextProps.type ) {
-			return;
-		}
-
-		this.request( nextProps );
-	}
+export default function QuerySiteDomains( { siteId, type } ) {
+	const requesting = useSelector( state => isRequestingPostCounts( state, siteId, type ) );
+	const dispatch = useDispatch();
+	const previousId = useRef( undefined );
+	const previousType = useRef( undefined );
 
-	request( props ) {
-		if ( props.requesting ) {
-			return;
+	useEffect( () => {
+		if ( ! requesting && siteId !== previousId.current && type !== previousType.current ) {
```

Comment:
Does storing the previous id in a ref look like the right approach for  comparing old and new id values?

Reply:
Answered elsewhere: it's best to avoid refs altogether, and figure out a way to have the effect dependency array perform the check.